viewing_party/party.py

```diff
@@ -1,23 +1,252 @@
+from statistics import multimode,mode
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    """Create function to store movie
+    Create empty dictionary to store the movie
+    Check for edge case of one item being none
+    """
+
+    movie = { "title": "", "genre" : "", "rating" : 0}
+    if title is None or genre is None or rating is None:
+        return None
+    else:
+        movie["title"]= title
+        movie["genre"]= genre
+        movie["rating"]= rating
+
+    return movie
+
+def add_to_watched(user_data, movie):
+    # Add movie a dictionary to watched
+    user_data["watched"].append(movie)
+    
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    # Add movie a dictionary to watchlist
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    """
+    Iterate through movies in user watchlist
+    Conditional if title is in user movie in watched
+    Then remove the watched movie from watchlist
+    """
+
+    for movie in user_data["watchlist"]:
+        if title in movie["title"]:
+            add_to_watched(user_data, movie) 
+            user_data["watchlist"].remove(movie)
+
+    return user_data
+    
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    """Set initial value of sum_rating to 0.
+    Create a new list of "watched" movies in user_data dict.
+    Iterate through movies in list of movies
+    Iterate through movies in list of movies
+    Find the sum of the integers in value for "rating" in the movie dict
+    If the input is an empty list, return 0.0 for average
+    Return the sum of movie ratings divided by the number of movies in input user_data dict
+    """
+
+    sum_rating = 0
+    
+    movie_list = user_data["watched"]
+
+    for movie in movie_list:
+        sum_rating += movie["rating"]
+    if len(movie_list) == 0:
+        return 0.0
+    else:
+        return sum_rating / len(movie_list)
+
+
+def get_most_watched_genre(user_data):
+    """
+    Create variable that is a list
+    Set variable to user's watched list
+    Loop through movies in user's watched list
+    Add movie genre to genre list
+    If genre list has 0 genres, then return None
+    Multimode returns most commonly occurring element in list
+
+    """
+    genre_list = []
+
+    movie_list = user_data["watched"]
+
+    for movie in movie_list:
+        genre_list.append(movie["genre"])
+
+    if len(genre_list) == 0:
+        return None
+    else:
+        return(multimode(genre_list)[0])

```

Comment:
Nice job finding a library function that handles this for you. I also love that you built it the long way by hand, too! It seems like your manual version isn't quite working (at least when I uncomment it and run it). You might benefit from coming back to this later for practice. :)